### ISSUE TITLE
local plugin resolution broken when plugin uses gatsby-browser.js but exposes exports via es6 `export`... 

### ISSUE THREAD

**Issue by busticated** [gatsbyjs/gatsby#4571]
i just noticed my local plugin was no longer being included in the `api-runner-browser.js` file shipped to the browser. digging in, i've found:

`v1.9.216` was the last good version of gatsby for me
as of `v1.9.218` my plugin was no longer included.

using `v1.9.233` (latest as i'm writing this) and tracing thru the code, i've found that the `resolve-module-exports.js` utility no longer recognizes my exported api methods ([see here](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/bootstrap/resolve-module-exports.js#L49))

my plugin is dead-simple

```javascript
export const onRouteUpdate = ({ location }) => {
	if (!window.analytics || typeof window.analytics.page !== 'function') {
		return;
	}

	window.analytics.page();
};
```

if i change the above to: `exports.onRouteUpdate` (instead of `export const onRouteUpdate`), it passes the api usage check and gets included in my build.

interestingly, `module.exports.onRouteUpdate` doesn't work either. 

seems like there's something funky in the babel transform and / or the code is enforcing incorrect assumptions about the transpiled output.

**Comment by m-allanson** [comment]
That's probably something from #3889, I'll do some digging.

**Comment by m-allanson** [comment]
A fix was released in version 1.9.235. @busticated thanks for the detailed bug report!
